### DISCUSSION TITLE
DNS in Git

### DISCUSSION THREAD

**Discussion by reesericci** [JupiterBroadcasting/jupiterbroadcasting.com#193] [Ideas]
I think it would be really cool if the jb.com DNS was managed transparently in a github repo.

**Comment by elreydetoda** [comment]
So, are you thinking something like Infrastructure as Code (IaC)? i.e. terraform, ansible, or pulumi?

**Reply by reesericci** [reply]
Yes but for specifically dns

**Reply by elreydetoda** [reply]
Ya, my only hesitation for that, with how JB currently does things, is it'd reveal the external IP address of the server that's hosting things.

If they'd be using things more IaC natively then there might not be any leaking of the IP address (through sanitization or authorization (someone how to be auth'ed to the CI)), then it'd be more feasible to not compromise on security for the convenience (IMO).

To clarify, this could be achievable because if they'd already used something like terraform to deploy servers and then they'd be able to reference a deployed server's external IP address (or reference with [cloudflare provider](https://developers.cloudflare.com/terraform/) (since that's [their DNS](https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/issues/16#issuecomment-1186580524)  )) through the terraform code instead of explicitly plugging in the IP address.

Although, I'd love to have this feature as well :grin: I love me some automation :grin:

**Reply by reesericci** [reply]
Isn't the server supposed to be public facing though?

**Reply by elreydetoda** [reply]
Ya, it's public facing (and currently they don't have the IP address masked), but when/if they inline cloudflare for DNS + extra features then the IP address _should_ be "masked" and no one _should_ be able to find it after that (with some caveats).

**Comment by gerbrent** [comment]
Conversation on this idea/issue moved to #258 
Locking this discussion to avoid duplication/divergence...
